src/validation/auth-validation.ts

```diff
@@ -0,0 +1,13 @@
+import { IUser } from 'src/models/user';
+import UnauthorizeError from '../errors/unauthorize-error';
+import { UserRoleEnum } from '../utils/common/enum';
+
+class AuthValidation {
+  public ensureLoggedIn(userRole: string | undefined, user: IUser | undefined) {
+    if (userRole === UserRoleEnum.GUEST || !user) {
```

Comment:
Think: how can userRole === ADMIN but user == null?

```suggestion
    const allowRoles = [UserRoleEnum.USER, UserRoleEnum.ADMIN];
    
    ...
    
    if (!allowRoles.includes(userRole)) {
```

The second param `user` can be removed because checking the user role is enough.